Transformer/UI/macOS/main.swift

```diff
@@ -0,0 +1,189 @@
+  DispatchQueue.global(qos: .background).async { body() }
+}
+
+class SwiftApplicationDelegate: NSObject, NSApplicationDelegate {
+  var window: NSWindow =
+      NSWindow(contentRect: NSRect(x: 0, y: 0, width: 648, height: 432),
+               styleMask: [.titled, .closable, .miniaturizable, .resizable],
+               backing: .buffered,
+               defer: false)
+
+  lazy var input: NSTextField =
+      NSTextField(frame: NSRect(x: 24, y: 392, width: 512, height: 20))
+  lazy var button: NSButton =
+      NSButton(frame: NSRect(x: 544, y: 386, width: 72, height: 32),
+               title: "Generate")
+  lazy var output: NSTextView =
+      NSTextView(frame: NSRect(x: 24, y: 128, width: 512, height: 256))
+  lazy var slider: NSSlider =
+      NSSlider(frame: NSRect(x: 24, y: 100, width: 512, height: 32))
+  lazy var label: NSLabel =
+      NSLabel(frame: NSRect(x: 24, y: 60, width: 512, height: 20),
+              title: "Loading Transformer ...")
+
+  var gpt: GPT2?
+
+  func applicationDidFinishLaunching(_ notification: Notification) {
+    self.slider.minValue = 0.0
+    self.slider.maxValue = 1.0
+    self.slider.value = 0.5
+
+    self.window.contentView?.addSubview(self.input)
+    self.window.contentView?.addSubview(self.button)
+
+    let view: NSScrollView = NSScrollView(frame: self.output.frame)
+    view.hasVerticalScroller = true
+    view.hasHorizontalScroller = true
+    view.documentView = self.output
+    self.output.textContainer?.widthTracksTextView = false
+    self.window.contentView?.addSubview(view)
+
+    // self.window.contentView?.addSubview(self.output)
+    self.window.contentView?.addSubview(self.slider)
+    self.window.contentView?.addSubview(self.label)
+
+    let ComicSansMS: NSFont = NSFont(name: "Comic Sans MS", size: 10)!
+
+    self.input.font = ComicSansMS
+    self.input.stringValue = "Introducing Swift for TensorFlow on macOS"
+
+    self.label.font = ComicSansMS
+    self.output.font = ComicSansMS
+    self.output.isEditable = false
+
+    self.button.target = self
+    self.button.action = #selector(generate)
+
+    onBackground {
+      do {
+        self.gpt = try GPT2()
+        onMain { self.label.text = "Transformer ready!" }
+      } catch {
+        onMain { self.label.text = "GPT2 Construction Failure" }
+      }
+    }
+
+    self.window.makeKeyAndOrderFront(nil)
+  }
+
+  func applicationShouldTerminateAfterLastWindowClosed(_ sender: NSApplication)
+      -> Bool {
+    return true
+  }
+}
+
+extension SwiftApplicationDelegate {
+  @objc
+  func generate() {
+    guard let gpt = self.gpt else { return }
+
+    output.text = input.text
+    if !input.text.isEmpty {
+      gpt.seed = gpt.embedding(for: input.text)
+    }
+    gpt.temperature = Float(self.slider.value)
+
+    onBackground {
+      for _ in 0 ..< 256 {
+        do {
+          let word: String = gpt.generate()
+          onMain {
+            try self.output.text = self.output.text + word
```

Comment:
```suggestion
            self.output.text = self.output.text + word
```
These two changes were required to align with how the GPT2 interface does error handling.